solr/core/src/java/org/apache/solr/search/grouping/distributed/shardresultserializer/SearchGroupsResultTransformer.java

```diff
@@ -85,7 +85,9 @@ public SearchGroupsResultTransformer(SolrIndexSearcher searcher) {
         searchGroup.groupValue = new BytesRef(groupValue);
       }
     }
-    searchGroup.sortValues = rawSearchGroupData.toArray(new Comparable[rawSearchGroupData.size()]);
+    @SuppressWarnings("unchecked")
+    Comparable<?>[] sv = (Comparable<?>[]) Array.newInstance(Comparable.class, rawSearchGroupData.size());
```

Comment:
I think you can do this, but I can't find a JLS reference for why it works.
```suggestion
    Comparable<?>[] sv = new Comparable<?>[rawSearchGroupData.size()];
```